streams/src/main/java/org/apache/kafka/streams/kstream/Materialized.java

```diff
@@ -69,7 +70,21 @@ public class Materialized<K, V, S extends StateStore> {
     // the built-in state store types
     public enum StoreType {
         ROCKS_DB,
-        IN_MEMORY
+        IN_MEMORY;
+
+        public static StoreType parse(final String storeType) {
+            switch (storeType) {
+                case StreamsConfig.IN_MEMORY:
+                    return StoreType.IN_MEMORY;
+                case StreamsConfig.ROCKS_DB:
+                default:
+                    // for backwards compatibility, we ignore invalid store
```

Comment:
Honestly I would just make this change now, and throw an exception if it's invalid. If you really want to keep the behavior the same for whatever reason, that's fine too, but we should at least log a warning

Reply:
👌